rust_src/src/lisp.rs

```diff
@@ -882,6 +882,12 @@ impl LispObject {
         )
     }
 
+    pub fn as_marker_or_error(self) -> LispMarkerRef {
+        self.as_marker().unwrap_or_else(
+            || wrong_type!(Qmarkerp, self),
```

Comment:
Yep. An example might be helpful here. The following snippets are not equivalent:

```rust
fn bar() -> u32 {
    println("bar called");
    42
}

foo.unwrap_or_else(|| bar()); // will print 'bar called' only when foo is None
foo.unwrap_or(bar()); // will always print 'bar called'
```